benchmarking/frameworks/caffe2/caffe2.py

```diff
@@ -284,46 +285,77 @@ def _collectData(self, total_num, output, results):
             # Android 5 has an issue that logcat -c does not clear the entry
             results = results[valid_run_idx[valid_runs-total_num]:]
         elif valid_runs < total_num:
-            if len(results) > prev_num:
+            if valid_runs > prev_num:
                 getLogger().info(
                         "%d items collected. Still missing %d runs. "
                         "Collect again." %
-                        (len(results) - prev_num, total_num - valid_runs))
-                return True
+                        (valid_runs - prev_num, total_num - valid_runs))
+                return valid_runs
             else:
                 getLogger().info(
                         "No new items collected, finish collecting...")
-        return False
+        return valid_runs
```

Comment:
You may return "total_num" to stop the loop.

Reply:
You are right, should return total_num

Reply:
You are right, should return total_num